### DISCUSSION TITLE
Problem Persists After #12121 Fix - How to Output WebP?(ComfyUI WORKFLOW tool)

### DISCUSSION THREAD

**Discussion by noriix** [langgenius/dify#12181] [Help] [answered]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

Dear @hjlarry, thank you for applying the fix for #12121. I'm still encountering issues, and have attached a screenshot. Perhaps there's a configuration I'm missing. Would you be able to tell me how to configure the output to WebP?
Please do not hesitate to inform me if you need any additional details.

### 2. Additional context or comments

![image](https://github.com/user-attachments/assets/9f21e0d2-662b-4bf0-ae1f-12fd62da4674)
![image](https://github.com/user-attachments/assets/fd175721-6e78-4e96-b429-2831a3db46c5)
![image](https://github.com/user-attachments/assets/6acd6818-4397-4511-b21b-099f01886415)

**Comment by hjlarry** [comment]
Hi, I also use the `SaveAnimatedWEBP` node for test, and not config anything else.
It seems strange,  the `mime_type` logic is based on the `filename` of the image,  can you print the filename to check this issue?

**Reply by noriix** [reply]
ComfyUI outputs with a .webp extension, but on the Dify side, it becomes .png. When I open the .png in an image viewer, it displays a message saying, "This is webp."

### ComfyUI
![image](https://github.com/user-attachments/assets/8cd988f1-32e7-4288-87a0-78abb0d017f8)

### Dify
![image](https://github.com/user-attachments/assets/7da881d2-74e8-4263-8cd8-8d4d5d855bf7)
![image](https://github.com/user-attachments/assets/45be7e2a-0b3e-467f-bad6-621e3783b1f1)

**Reply by hjlarry** [reply]
can you debug this line,  to check what the filename get from comfy?
https://github.com/langgenius/dify/blob/ead4b341277108bbb42aa340490b1a6c76d7604b/api/core/tools/provider/builtin/comfyui/tools/comfyui_client.py#L128

**Reply by noriix** [reply]
I'm not really sure how to debug, but I hope this information is what you need.

{
  "text": "",
  "files": [
    {
      "dify_model_identity": "__dify__file__",
      "id": null,
      "tenant_id": "d282ce03-ab7a-44cf-a986-7dac768d2073",
      "type": "image",
      "transfer_method": "tool_file",
      "remote_url": null,
      "related_id": "60094558-635c-4df2-b3af-b881dc47e8e2",
      "filename": "a433be71fb154c6e966423a8297544ea.png",
      "extension": ".png",
      "mime_type": "image/png",
      "size": 268394,
      "url": "/files/tools/60094558-635c-4df2-b3af-b881dc47e8e2.png?timestamp=1735381452&nonce=b8ff0e3d16cc4af7bdc97cfd19a61c61&sign=4-yMfjvTzXFvGIzvcwdCSux1Zc7SETtyDgGFv-dhNFE="
    }
  ],
  "json": []
}
![image](https://github.com/user-attachments/assets/48a6900b-b01a-4d97-b21c-ead25e717bc8)

**Reply by hjlarry** [reply]
are you use the docker version of dify?

**Reply by noriix** [reply]
Yes, I'm using docker version.

**Reply by hjlarry** [reply, marked as answer]
oh, the docker version of dify is only updated when offical team publish a new version, it's no matter with your local code or repo change.  So wait for the new version please.

**Reply by noriix** [reply]
Oh, that makes sense. git pull won't apply to the Docker version then. I apologize for troubling you during this busy end-of-year period.

**Reply by hjlarry** [reply]
It doesn't matter.  I'm always happy to help :)
